examples/temperature-measurement-app/esp32/main/diagnostic-logs-provider-delegate-impl.cpp

```diff
@@ -278,11 +307,23 @@ CHIP_ERROR LogProvider::StartLogCollection(IntentEnum intent, LogSessionHandle &
 
 CHIP_ERROR LogProvider::EndLogCollection(LogSessionHandle sessionHandle, CHIP_ERROR error)
 {
-    if (error != CHIP_NO_ERROR)
+#ifdef CONFIG_ENABLE_ESP_DIAGNOSTICS_TRACE
+    VerifyOrReturnError(mStorageInstance != nullptr, CHIP_ERROR_INTERNAL,
+                        ChipLogError(DeviceLayer, "Diagnostic Storage instance cannot be null."));
+    if (error == CHIP_NO_ERROR)
     {
-        // Handle the error
-        ChipLogProgress(DeviceLayer, "End log collection reason: %s", ErrorStr(error));
+        CHIP_ERROR err = mStorageInstance->ClearReadMemory(sReadEntries);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(DeviceLayer, "Failed to clear diagnostic read entries");
+        }
+        else
+        {
+            ChipLogProgress(DeviceLayer, "Cleared all read diagnostics successfully");
```

Comment:
This will be printed after every call to `EndLogCollection()`, if this is too much of logs, may be can we make it `ChipLogDetail()`?